torch/_logging/_internal.py

```diff
@@ -0,0 +1,424 @@
+                    )
+                if val != DEFAULT_LOG_LEVEL:
+                    log_state.enable_log(
+                        log_registry.log_alias_to_log_qname[alias], val
+                    )
+            else:
+                # Check if it is a qualified name log
+                # if so, check that its root logger is parent
+                # if so set its level appropriately
+                # if not, register it? (maybe)
+                raise ValueError(
+                    f"Unrecognized log or artifact name passed to set_logs: {alias}"
+                )
+
+        _init_logs()
+
+    _set_logs(
+        dynamo=dynamo,
+        aot=aot,
+        inductor=inductor,
+        bytecode=bytecode,
+        aot_forward_graph=aot_forward_graph,
+        aot_backward_graph=aot_backward_graph,
+        aot_joint_graph=aot_joint_graph,
+        graph=graph,
+        graph_code=graph_code,
+        guards=guards,
+        output_code=output_code,
+        schedule=schedule,
+    )
+
+
+def register_log(setting_name, log_name):
+    """
+    Enables a log to be controlled by the env var and user API with the setting_name
+    Args:
+        setting_name:  the shorthand name used in the env var and user API
+        log_name:  the log name that the setting_name is associated with
+    """
+    log_registry.register_log(setting_name, log_name)
+
+
+def register_artifact(setting_name, off_by_default=False):
+    """
+    Enables an artifact to be controlled by the env var and user API with name
+    Args:
+        setting_name: the shorthand name used in the env var and user API
+        off_by_default: whether this artifact should be logged when the ancestor loggers
+            are enabled at level DEBUG
+    """
+    log_registry.register_artifact_name(setting_name, off_by_default)
+
+
+def getArtifactLogger(module_qname, artifact_name):
+    if artifact_name not in log_registry.artifact_names:
+        raise ValueError(
+            f"Artifact name: {artifact_name} not registered,"
+            f"please call register_artifact({artifact_name}) in torch._logging.registrations."
+        )
+    qname = module_qname + f".__{artifact_name}"
+    log = logging.getLogger(module_qname + f".__{artifact_name}")
+    log.artifact_name = artifact_name  # type: ignore[attr-defined]
+    log_registry.register_artifact_log(qname)
+    configure_artifact_log(log)
+    return log
+
+
+INCR_VERBOSITY_CHAR = "+"
+DECR_VERBOSITY_CHAR = "-"
+VERBOSITY_REGEX = (
+    "("
+    + "|".join([re.escape(INCR_VERBOSITY_CHAR), re.escape(DECR_VERBOSITY_CHAR)])
+    + "?)"
+)
+
+
+def configure_artifact_log(log):
+    # if parent log is set to debug, but this artifact is off by default
+    # set propagate to False so that this artifact is not propagated
+    # to its ancestor logger
+    # this artifact is only logged when explicitly enabled (occurs below)
+    if (
+        log_registry.is_off_by_default(log.artifact_name)
```

Comment:
Hmph, I'm not convinced we actually need this functionality, but OK.

Reply:
I mean, we don't but, printing the output code all the time is pretty annoying, there are only specific situations when people want to see it, or else it will take up the whole log! haha

Reply:
I guess I expect DEBUG to take up pages anyway, but sure, s'fine.